tools/contrib/fsurdat_modifier.py

```diff
@@ -0,0 +1,356 @@
+            datefmt='%y-%m-%d %H:%M:%S')
+
+    handler.setFormatter(fmt)
+    root_logger.addHandler(handler)
+
+    # setup logging to console
+    stream_handler = logging.StreamHandler(sys.stdout)
+    stream_handler.setFormatter(fmt)
+    root_logger.addHandler(stream_handler)
+
+    # redirect stdout/err to log file
+    StreamToLogger.setup_stdout()
+    StreamToLogger.setup_stderr()
+
+
+class StreamToLogger(object):
+    """
+    Custom class to log all stdout and stderr streams.
+    modified from:
+    https://www.electricmonk.nl/log/2011/08/14/redirect-stdout-and-stderr-to-a-logger-in-python/
+    """
+    def __init__(self, stream, logger, log_level=logging.INFO,
+                 also_log_to_stream=False):
+        self.logger = logger
+        self.stream = stream
+        self.log_level = log_level
+        self.linebuf = ''
+        self.also_log_to_stream = also_log_to_stream
+
+    @classmethod
+    def setup_stdout(cls, also_log_to_stream=True):
+        """
+        Setup logger for stdout
+        """
+        stdout_logger = logging.getLogger('STDOUT')
+        sl = StreamToLogger(sys.stdout, stdout_logger, logging.INFO, also_log_to_stream)
+        sys.stdout = sl
+
+    @classmethod
+    def setup_stderr(cls, also_log_to_stream=True):
+        """
+        Setup logger for stdout
+        """
+        stderr_logger = logging.getLogger('STDERR')
+        sl = StreamToLogger(sys.stderr, stderr_logger, logging.ERROR, also_log_to_stream)
+        sys.stderr = sl
+
+    def write(self, buf):
+        temp_linebuf = self.linebuf + buf 
+        self.linebuf = ''
+        for line in temp_linebuf.splitlines(True):
+            if line[-1] == '\n':
+                self.logger.log(self.log_level, line.rstrip())
+            else:
+                self.linebuf += line
+
+    def flush(self):
+        if self.linebuf != '': 
+            self.logger.log(self.log_level, self.linebuf.rstrip()) 
+        self.linebuf = ''
+
+
+def main ():
+
+    # Parse arguments from the command line
+    args = get_parser().parse_args()
+
+    # Set up logging
+    today = date.today()
+    today_string = today.strftime("%Y%m%d")
+    pwd = os.getcwd()
+    log_file = os.path.join(pwd, today_string + '.log')
+    log_level = logging.DEBUG
+    setup_logging(log_file, log_level)
+    log = logging.getLogger(__name__)
+
+    print("User = " + myname)
+    print("Current directory = " + pwd)
+
+    #--  Modify landunit structure
+    overwrite_single_pft = args.overwrite_single_pft
+    dominant_pft         = args.dom_pft
+    zero_nonveg_landunits= args.zero_nonveg
+    uniform_snowpack     = args.uni_snow
+    no_saturation_excess = args.no_saturation_excess
+
+    #--  Create ModifyFsurdat Object
+    modify_fsurdat = ModifyFsurdat(overwrite_single_pft, dominant_pft,
+                                   zero_nonveg_landunits, uniform_snowpack,
+                                   no_saturation_excess)
```

Comment:
- [ ] Given that this argument list is likely to become long, I'd suggest:
  - Not bothering making local copies of the args, but instead passing things like args.uni_snow directly
  - Using argument keywords in the function call to avoid accidentally associating the wrong arguments